code/controller/solenoid.cpp

```diff
@@ -0,0 +1,74 @@
+#include "solenoid.h"
+#include <ostream>
+
+Solenoid::Solenoid(const QString& serial_port): Controller(1, 1) {
+    // TODO: change actuator setup to be used for general serial set up
+    if ( serial_port.isEmpty() ) {
+        // TODO: provide error if this does not succeed
+        stream.open("/dev/ttyACM0");        // manually changed to match port indicated in arduino IDE
```

Comment:
it should be possible to autodetect ports if not provided

Reply:
We currently have that set up in `actuatorsetup.cpp` but we learned that it doesn't properly detect the arduino USB port.